route/v1/stable.route.testsuite.yaml

```diff
@@ -20,3 +20,383 @@ tests:
+  - name: Should not allow to set/delete dynamic response header values with not allowed sample fetchers.
+    initial: |
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-edge-not-allowed-values
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der1,base64]"
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are res.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should not allow to set/delete dynamic response header values with not allowed converters.
+    initial: |
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-edge-not-allowed-values
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,bogus]"
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are res.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should not allow to set/delete dynamic response header values containing req.hdr fetcher.
```

Comment:
Do we need an equivalent for the request header values containing `res.hdr` fetcher?

Reply:
not needed as we have a regex for request /req/ and /ssl_c_der/  so it does not allow any other fetcher while providing a dynamic value. Earlier i had provided these test because I had written a CEL for reponse array and Value had pattern

Reply:
Yes but now you have changed the regexes, I think it's valid to require a test case that shows you can't put `res.hdr` into the request headers, please add